riscv/src/bootloader.rs

```diff
@@ -78,6 +91,48 @@ P0, P1, P2, P3 <== poseidon_gl(P0, P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11)
         }
     }
 
+    bootloader.push_str(
+        r#"
+// Simulate a Merkle proof, using 0 as the sibling hashes for now...
+// This is an unrolled loop that for each level:
+// - If the ith bit of the page number is 0:
+//   - Load sibling into registers P4-P7
+// - Else:
+//   - Write registers P0-P3 to registers P4-P7
+//   - Load sibling into registers P0-P3
+// - Hash registers P0-P11, storing the result in P0-P3
+//
+// At the end of the loop, we'll have the Merkle root in P0-P3.
+"#,
+    );
+
+    for i in 0..merkle_tree_depth {
+        let mask = 1 << i;
+        bootloader.push_str(&format!(
+            r#"
+x4 <== and(x3, {mask});
```

Comment:
@leonardoalt Then I'd rather try to get rid of this, because here it get's called 22 times per page (once for each level in the Merkle tree). But still sounds fine to me (e.g. 220 calls = 880 rows for 10 pages, which seems to be a typical amount).

Reply:
yea let's keep it for now, no need to over-optimize it at this point